### PR TITLE
Fix helm chart when containerMode.type=dind.

### PR DESCRIPTION
**Pull request by TingluoHuang**
The chart is not rendered properly when `--set containerMode.type=dind`

It was rendered to:
```
     containers:
      - name: runner
        command: [/actions-runner/run.sh]
        image: ghcr.io/actions/actions-runner:latest
        env:
        volumeMounts:
          - name: work
            mountPath: /actions-runner/_work
          - name: dind-cert
            mountPath: /certs/client
```

Notice the `env` section is empty due to the bug.

### CODE REVIEW

File: charts/auto-scaling-runner-set/templates/_helpers.tpl [hunk 1]
```diff
@@ -208,18 +208,18 @@ env:
           {{- end }}
```
Comment (TingluoHuang) [inline charts/auto-scaling-runner-set/templates/_helpers.tpl:208]: hide whitespace when reviewing.

TLDR, the bug is `{{- end }}` get added in the wrong line.

**Review by rentziass — APPROVED** [review]
